src/mscorlib/src/System/Reflection/Emit/CustomAttributeBuilder.cs

```diff
@@ -150,12 +156,26 @@ private bool ValidateType(Type t)
             // Now verify that the types of the actual parameters are compatible with the types of the formal parameters.
             for (i = 0; i < paramTypes.Length; i++)
             {
-                if (constructorArgs[i] == null)
+                object constructorArg = constructorArgs[i];
+                if (constructorArg == null)
+                {
+                    if (paramTypes[i].IsPrimitive)
+                    {
+                        throw new ArgumentNullException("constructorArgs[" + i + "]");
+                    }
                     continue;
+                }
+                Type constructorArgType = constructorArg.GetType();
                 TypeCode paramTC = Type.GetTypeCode(paramTypes[i]);
-                if (paramTC != Type.GetTypeCode(constructorArgs[i].GetType()))
-                    if (paramTC != TypeCode.Object || !ValidateType(constructorArgs[i].GetType())) 
+                if (paramTC != Type.GetTypeCode(constructorArgType))
+                {
+                    if (paramTC != TypeCode.Object || !ValidateType(constructorArgType))
                         throw new ArgumentException(Environment.GetResourceString("Argument_BadParameterTypeForConstructor", i));
+                }
+                if (constructorArgType == typeof(IntPtr) || constructorArgType == typeof(UIntPtr))
+                {
+                    throw new ArgumentException(Environment.GetResourceString("Argument_BadParameterTypeForConstructor", i), "constructorArgs[" + i + "]");
```

Comment:
Is this "throw" reachable now? You already called ValidateType() on the formal type, and called ValidatedType on the actual type if the actual type's TC differed from the formals. One of those should have already thrown the "can't use an IntPtr" exception.